### DISCUSSION TITLE
1.6 高阶函数

### DISCUSSION THREAD

**Discussion by giscus** [csfive/composing-programs-zh#104] [Announcements]
# 1.6 高阶函数

SICP Python 中文翻译

https://composingprograms.netlify.app/1/6

**Comment by CharlieCRX** [comment]
这一节要是没翻译，我可能就死在这了🥲

**Reply by xiaobai-first** [reply]
有翻译好像也要死了,我得去看看视频了

**Comment by Harbour-z** [comment]
感谢翻译！这一节用中文看都断断续续看了几天，让人头大

**Comment by jahox0102** [comment]
感谢大佬们的翻译！

**Comment by yangjdyc** [comment]
感谢翻译！作为初学者还是不太理解匿名函数的用法，头大啊

**Comment by WUADN** [comment]
1.6.3 python tutor 下面第三段：
这个求值过程中最关键的部分是将 sqrt_update 的父环境变成了通过调用 sqrt_update 创建的（局部）帧。此帧还带有 [parent=f1] 的注释。
原文“The most critical part of this evaluation procedure is the transfer of the parent for sqrt_update to the frame created by calling sqrt_update. This frame is also annotated with [parent=f1].”
应该翻译为“这个求值过程其中最关键的部分是将 变量的值从该函数的父环境传递到调用sqrt_update创建的帧中 ，并且在执行环境中包含了一个 parent 变量，其值被设置为 f1”这个应该更符合原文

**Reply by WUADN** [reply]
“并且在执行环境中包含了一个 parent 变量，其值被设置为 f1”：标记parent=f1

**Comment by kaizjh** [comment]
感谢大佬们的翻译，提一个小建议：
我在这一节第一次用了一下深色模式，然后发现深色模式下的 visualize 环境图以及其他截图都是浅色的（白色），一下深色一下浅色，看的时候有点刺眼，如果大佬们能把这个问题优化一下就再好不过了。
再次感谢各位的辛苦付出。

**Comment by SDU-YueQiu** [comment]
醍醐灌顶了

**Comment by enternal111** [comment]
牛顿法没有处理导数为0的情况

**Reply by enternal111** [reply]
以及有多个解的情况

**Comment by JensenJin19** [comment]
这一节好难，学1.1~1.5的时候只有frame困扰了自己一会儿，到了这节难度明显上来了

**Comment by zgozh** [comment]
怎么第一章就上强度了

**Comment by huahai27** [comment]
打卡
感谢大佬的翻译，这节真的硬核

**Comment by Homovexillues** [comment]
强烈建议这里如果有疑惑的地方就找GPT-o1讲一下自己的想法和理解，或者询问他的理解，效果很好

**Comment by KyoSekai** [comment]
>>>这玩意影响阅读啊

**Comment by SwordKillCowardice** [comment]
这一节有点小难

**Comment by MasterpieceXu** [comment]
作为返回值的函数其实是叫“闭包”，起码chapgpt是这么说的，在lab2和homework2中都运用了1.6.4，反而1.6.5在slide中好像属于optional的内容，久违的泰勒展开，我都忘记了……

**Comment by swxinezz** [comment]
看到大家都说难我就放心了 我还以为我是🐖

**Reply by sxc-323** [reply]
哈哈哈，我也觉得好难

**Comment by RisonaSuki** [comment]
这里需要理解牛顿法的原理，教材讲的不是很好，容易云里雾里。
牛顿法事实上利用了泰勒展开式

**Reply by swxinezz** [reply]
对的，有点太偏向应用导致学起来有点抽象了

**Comment by CrazyMaster001** [comment]
感谢各位佬，mol了
